### ISSUE TITLE
Add transformer sequence model for predicting next token with l2 loss

### ISSUE THREAD

**Issue by acmiyaguchi** [dsgt-birdclef/birdclef-2023#67]
We will want to start work on building a sequence model -- this hopefully works better than a naive model that's been retrained for the birdnet task.

**Comment by acmiyaguchi** [comment]
I have some transformer code from another project if you want something to shim from -- this uses pytorch-lightning for dealing with stuff. This mostly pulls from pytorch tutorials, so this should look somewhat familiar:

https://github.com/acmiyaguchi/amex-default-prediction/blob/main/amex_default_prediction/torch/net.py

There are couple of tests, one that I think might be useful is to create a datamodule and check that it feeds through the transformer okay:

https://github.com/acmiyaguchi/amex-default-prediction/blob/211358b3232d1be924fad3d486d3bf2d64f2683d/tests/test_torch.py#L232-L259

We have some test fixtures that should work fine for testing things in isolation:

https://github.com/dsgt-birdclef/birdclef-2023/blob/ead34674e49ef9c4fa35fe6e447d241ddd98d897/tests/conftest.py#L13-L28

**Comment by ChrisHayduk** [comment]
After getting transfomer trained, do the following:

- Compare to baselines (autoregressive model with context of 1 previous token, LSTM model in issue #68)
- Add classifier head and compare performance to baseline classifier
- (Time permitting) train the transformer against 2021/2022 dataset to see if it helps with model performance/generalization
